storage/google/cloud/storage/_helpers.py

```diff
@@ -126,6 +126,8 @@ def _set_properties(self, value):
         self._properties = value
         # If the values are reset, the changes must as well.
         self._changes = set()
+        if hasattr(self, '_label_removals'):
+            self._label_removals.clear()
```

Comment:
Maybe this is over-picky, but ISTM it make more sense to override `_set_properties` and do this work there.